evennia/contrib/game_systems/buffs/tests.py

```diff
@@ -0,0 +1,366 @@
+# in a module tests.py somewhere i your game dir
+from unittest.mock import Mock, patch
+from evennia import DefaultObject, create_object
+from evennia.utils import create
+from evennia.utils.utils import lazy_property
+from .samplebuffs import StatBuff
+
+# the function we want to test
+from .buff import BaseBuff, Mod, BuffHandler, BuffableProperty
+from evennia.utils.test_resources import EvenniaTest
+
+
+class _EmptyBuff(BaseBuff):
+    pass
+
+
+class _TestModBuff(BaseBuff):
+    key = "tmb"
+    name = "tmb"
+    flavor = "modderbuff"
+    maxstacks = 5
+    mods = [Mod("stat1", "add", 10, 5), Mod("stat2", "mult", 0.5)]
+
+
+class _TestModBuff2(BaseBuff):
+    key = "tmb2"
+    name = "tmb2"
+    flavor = "modderbuff2"
+    maxstacks = 1
+    mods = [Mod("stat1", "mult", 1.0), Mod("stat1", "add", 10)]
+
+
+class _TestTrigBuff(BaseBuff):
+    key = "ttb"
+    name = "ttb"
+    flavor = "triggerbuff"
+    triggers = ["test1", "test2"]
+
+    def at_trigger(self, trigger: str, *args, **kwargs):
+        if trigger == "test1":
+            self.owner.db.triggertest1 = True
+        if trigger == "test2":
+            self.owner.db.triggertest2 = True
+
+
+class _TestConBuff(BaseBuff):
+    key = "tcb"
+    name = "tcb"
+    flavor = "condbuff"
+    triggers = ["condtest"]
+
+    def conditional(self, *args, **kwargs):
+        return self.owner.db.cond1
+
+    def at_trigger(self, trigger: str, attacker=None, defender=None, damage=0, *args, **kwargs):
+        defender.db.att, defender.db.dmg = attacker, damage
+
+
+class _TestComplexBuff(BaseBuff):
+    key = "tcomb"
+    name = "complex"
+    flavor = "combuff"
+    triggers = ["comtest", "complextest"]
+
+    mods = [
+        Mod("com1", "add", 0, 10),
+        Mod("com1", "add", 15),
+        Mod("com1", "mult", 2.0),
+        Mod("com2", "add", 100),
+    ]
+
+    def conditional(self, cond=False, *args, **kwargs):
+        return not cond
+
+    def at_trigger(self, trigger: str, *args, **kwargs):
+        if trigger == "comtest":
+            self.owner.db.comtext = {"cond": True}
+        else:
+            self.owner.db.comtext = {}
+
+
+class _TestTimeBuff(BaseBuff):
+    key = "ttib"
+    name = "ttib"
+    flavor = "timerbuff"
+    maxstacks = 1
+    tickrate = 1
+    duration = 5
+    mods = [Mod("timetest", "add", 665)]
+
+    def at_tick(self, initial=True, *args, **kwargs):
+        self.owner.db.ticktest = True
```

Comment:
I would expect this to mock out the actual `delay` call. We don't want timers going off inside the unit test suite.